browser_use/browser/browser.py

```diff
@@ -246,24 +224,21 @@ async def _setup_browser(self, playwright: Playwright) -> PlaywrightBrowser:
 	async def close(self):
 		"""Close the browser instance"""
 		try:
-			if not self.config._force_keep_browser_alive:
-				if self.playwright_browser:
-					await self.playwright_browser.close()
-					del self.playwright_browser
-				if self.playwright:
-					await self.playwright.stop()
-					del self.playwright
-
+			if self.config._force_keep_browser_alive:
+				return
+			if self.playwright_browser:
+				await self.playwright_browser.close()
+				self.playwright_browser = None
+			if self.playwright:
+				await self.playwright.stop()
+				self.playwright = None
 		except Exception as e:
 			logger.debug(f'Failed to close browser properly: {e}')
 		finally:
-			self.playwright_browser = None
-			self.playwright = None
-
 			gc.collect()
```

Comment:
I am not sure if the earlier one has a correct behaviour. The playwright_browser and playwright is set to None irrespective of `_force_keep_browser_alive`.

Reply:
I believe the existing behavior is correct, it releases the playwright channel memory but does not close the browser, which is what's expected if the user sets `_force_keep_browser_alive`.

Reply:
updated.